### ISSUE TITLE
dxvk 1.2.2 World of Tanks always starts in windowed mode

### ISSUE THREAD

**Issue by Kevinmmi** [doitsujin/dxvk#1096]
Hi,
Firstly thanks for the great work :-)

Since updating to v1.2.2 WoT will always start in a windowed mode, if I try and change the option in the game from windowed to full-screen it just drops back to windowed if I first change to windowed borderless then to full-screen it will take the settings however if I exit the game and go back in it will started windowed again.

I haven't captured the apitrace at this time in case it's something dumb I'm doing but can do if required.
If I go back to 1.2.1 everything is fine after the first login to game and set full-screen.

So I have reverted for now to 1.2.1.

### Software information
World Of Tanks

### System information
- GPU: Nvidia
- Driver: 418.56
- Wine version: wine-4.10 (Staging)
- DXVK version: 1.2.2

Regards

Kevin

**Comment by doitsujin** [comment]
Can you **please** post the log files that were requested in the issue template? When you do, set `DXVK_LOG_LEVEL=debug` before starting the game.

I can't reproduce the problem.

**Comment by Kevinmmi** [comment]
Hi,
Apologies below are the links to all 3 files

https://www.dropbox.com/s/ku74cg0fsvkq9h5/WorldOfTanks_d3d11.log.zip?dl=0
https://www.dropbox.com/s/nonihhqucn3xnv9/WorldOfTanks_dxgi.log.zip?dl=0
https://www.dropbox.com/s/sos516yn0m3nhw6/WoT.trace.zip?dl=0

Regards

Kevin

**Comment by doitsujin** [comment]
Your DXGI log shows DXVK v0.71, something might have gone wrong there.

Nothing suspicious in the apitrace so far, although I see you have a high refresh rate monitor which is known to sometimes cause issues on wine (but then again, the display modes are enumerated correctly).

**Comment by Kevinmmi** [comment]
Hi odd about the DXVK version as Ive enabled the HUD and can see the right 1.2.2 version in the game.
As mentioned if I go back to 1.2.1 , all other settings  the same, the issue goes away.

Regards

Kevin

**Comment by Sigtechnica** [comment]
I have a similar issue with World Of Warships, in the weekend it was still fine. 
Today I wanted to play, but noticed first that gsync was not enabled, and noticed the game was running in full  screen windowed... 
changed it to fullscreen in the game it seem to work. 

I also noticed that my mouse cursor became a black square. Thought ok no biggy. But when playing some of the textures / shaders went fubar, inverted or wrongly mapped , etc... 

First I blamed the wine-staging update. rolled back to wine-staging 4.8, but still same issue .
Cleared the mesa cache folder , then the WorldOfWarships64.dxvk-cache file

Then I rolled back to 1.2.1 with wine-staging 4.8 and it worked again, then upgrade I wine-staging back again to the latest , still working. So something is breaking it, tried Diablo 3 it look like it was working fine

https://www.dropbox.com/s/hzgcv9324gq6fgu/World_of_Warships_EU_dxvk_1.2.1.zip?dl=0
https://www.dropbox.com/s/7xdqk0d0ugrxe84/World_of_Warships_EU_dxvk_1.2.2.zip?dl=0

haven't got the time yet todo the stack trace will do it tomorrow

**Comment by doitsujin** [comment]
@Sigtechnica World of Warships works fine with 1.2.2 on my end. "textures / shaders went fubar" is also not a very helpful description of your problem, can you post a screenshot or something?

There were basically no changes between 1.2.1 and 1.2.2 that would affect textures, so I don't really understand why that would break anything.

**Comment by Sigtechnica** [comment]
https://www.dropbox.com/s/tnutvq8ztbirjx7/IMG_20190625_004836.jpg?dl=0
https://www.dropbox.com/s/6rfxg54brjl4mio/IMG_20190625_004858.jpg?dl=0
https://www.dropbox.com/s/sxm0oshosamozmv/IMG_20190625_004932.jpg?dl=0

**Comment by doitsujin** [comment]
Yeah, no, that definitely doesn't happen for me.

**Comment by doitsujin** [comment]
Would you be able to bisect the issue? Since I cannot reproduce it at all, I won't be able to do much without at least knowing what caused it.

**Comment by Sigtechnica** [comment]
yes, never used git bisect (never needed it :P ) , so cool thing to try. When I got some time, I will start digging.

**Comment by doitsujin** [comment]
No need, I just figured out what broke it. Thing is, I have no idea **why**, so it might take a while until this is fixed.

**Comment by doitsujin** [comment]
@Sigtechnica can you test if latest `master` fixes your World of Warships issues? Here's a build:
[dxvk-master.tar.gz](https://github.com/doitsujin/dxvk/files/3325253/dxvk-master.tar.gz)

**Comment by Sigtechnica** [comment]
Tested it and it works , many thanks :) . didn't check my mail / messages here, and played a little  with the bisect tool, found that The commit causing the issue is : https://github.com/doitsujin/dxvk/commit/3168626f4be39672ec2ba626ed6c26e22a8f9c62 . 

cool fix btw :)

**Comment by Kevinmmi** [comment]
Hi,

Ok so I have updated to 1.2.3 and still have the issue that the game starts windowed and then I have to change the settings in the following order before I can play full screen?
Starts in "windowed" mode
I have to set "Windowed (borderless)" then save
I then can set "full-screen" and save.

The game plays fine but as soon as I exit and restart I have to go through the same process for getting full screen back, not an issue if I'm running 1.2.1?

Regards

Kevin

**Comment by doitsujin** [comment]
As mentioned multiple times, I cannot reproduce your problem.

Can you bisect this on your end?

**Comment by Kevinmmi** [comment]
Hi,

Apologies for being dumb but what do you mean by "bisect"?
Is this some other documented method for analyzing the issue?

Kevin

**Comment by doitsujin** [comment]
https://git-scm.com/docs/git-bisect

**Comment by Sigtechnica** [comment]
Try to upgrade nvidia drivers to 430.26 
I had a similar issue, in world of warships, was always starting in windowed mode full screen, thus not triggering Gsync , so put it manual to full screen a few times .
I also removed and then reinstalled the game
played in winecfg with unchecking the "Allow the window manager to Decorate the windows" aswell the control the windows.

**Comment by doitsujin** [comment]
Are any of you guys **not** using Gnome or Cinnamon?

**Comment by Kevinmmi** [comment]
OI', using Linux Mint 18 but the KDE version.

**Comment by Kevinmmi** [comment]
@Sigtechnica,

I have tried the upgrade a few times before but had issues with it however I tried the last option you mentioned unchecking allow windows to decorate and that appears to have sorted the issue.
Not exactly sure why this would be the case between dxvk versions but I'm not going to worry too much as it now starts full-screen on the latest 1.2.3 version.
Thanks

**Comment by Sigtechnica** [comment]
funny, because I since yesterday it started happening with me again, but I am not sure it is related to dxvk, with diablo 3 for example I don't have that issue. 
I am using XFCE , ubuntu 19.04 , because it was the easiest way to use gsync.

**Comment by Blisto91** [comment]
I tried launching this game in wine 7.6 with a nvidia 960 with 510 driver and dxvk 1.2.2 on both X11 kde and wayland. I cant seem to reproduce the issue.
Do any of you still have the problem?

**Comment by Blisto91** [comment]
@Kevinmmi @Sigtechnica Friendly ping.

**Comment by Sigtechnica** [comment]
lets say 99% of the time it works now without starting in window mode.
I was thinking it could be related after an update or something

**Comment by Blisto91** [comment]
Hmm okay. I'll take that as it is still an issue from time to time.

**Comment by Blisto91** [comment]
@Sigtechnica Sorry for the periodic necro pings.
Do you know if anything have changed on this front?

**Comment by Blisto91** [comment]
@Sigtechnica @Kevinmmi Friendly ping

**Comment by Sigtechnica** [comment]
just tried it, I have not played many wargaming lately, but issue is still there.
maybe I should a fresh wine install, because now sometimes when switching to full screen , the screen stays black ... :)
